public/scripts/popup.js

```diff
@@ -260,28 +261,52 @@ export class Popup {
-            inputElement.type = 'checkbox';
-            inputElement.id = input.id;
-            inputElement.checked = input.defaultState ?? false;
-            label.appendChild(inputElement);
-            const labelText = document.createElement('span');
-            labelText.innerText = input.label;
-            labelText.dataset.i18n = input.label;
-            label.appendChild(labelText);
-
-            if (input.tooltip) {
-                const tooltip = document.createElement('div');
-                tooltip.classList.add('fa-solid', 'fa-circle-info', 'opacity50p');
-                tooltip.title = input.tooltip;
-                tooltip.dataset.i18n = '[title]' + input.tooltip;
-                label.appendChild(tooltip);
+            if (!input.type || input.type === 'checkbox') {
+                const label = document.createElement('label');
+                label.classList.add('checkbox_label', 'justifyCenter');
+                label.setAttribute('for', input.id);
+                const inputElement = document.createElement('input');
+                inputElement.type = 'checkbox';
+                inputElement.id = input.id;
+                inputElement.checked = Boolean(input.defaultState ?? false);
+                label.appendChild(inputElement);
+                const labelText = document.createElement('span');
+                labelText.innerText = input.label;
+                labelText.dataset.i18n = input.label;
+                label.appendChild(labelText);
+
+                if (input.tooltip) {
+                    const tooltip = document.createElement('div');
+                    tooltip.classList.add('fa-solid', 'fa-circle-info', 'opacity50p');
+                    tooltip.title = input.tooltip;
+                    tooltip.dataset.i18n = '[title]' + input.tooltip;
+                    label.appendChild(tooltip);
+                }
+
+                this.inputControls.appendChild(label);
             }
 
-            this.inputControls.appendChild(label);
+            if (input.type === 'text') {
+                const label = document.createElement('label');
+                label.classList.add('text_label', 'justifyCenter');
+                label.setAttribute('for', input.id);
+
+                const inputElement = document.createElement('input');
+                inputElement.classList.add('text_pole');
+                inputElement.type = 'text';
+                inputElement.id = input.id;
+                inputElement.value = String(input.defaultState ?? '');
+                inputElement.placeholder = input.tooltip ?? '';
+
+                const labelText = document.createElement('span');
+                labelText.innerText = input.label;
+                labelText.dataset.i18n = input.label;
+
+                label.appendChild(labelText);
+                label.appendChild(inputElement);
+
+                this.inputControls.appendChild(label);
+            }
```

Comment:
We should throw on invalid input type, or... provide a default case, or... at least log a warning.

Funny idea.
Why not fire an event, so custom extensions could *build* their own input types for popups.
Not that they couldn't do it manually in the popup body too, but.. dunno. If we have an "unknown" block, dunno.
Otherwise, I would personally prefer an error throw. Because whoever coded the popup made a mistake in code.